From 142f895ce9a6631bb874b93052a2d3ff6c7747ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 7 Jun 2017 17:27:43 +0200
Subject: [PATCH] Undefine BOM_UTF8
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Work around BOM_UTF8 clash between wxGTK-2.8.18 and perl-5.26.0.

CPAN RT#121464
http://trac.wxwidgets.org/ticket/13599

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 cpp/wxapi.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cpp/wxapi.h b/cpp/wxapi.h
index bdee14f..c0a2924 100644
--- a/cpp/wxapi.h
+++ b/cpp/wxapi.h
@@ -55,6 +55,12 @@ inline FILE* _wxPli_stderr() { return stderr; }
 WXPL_EXTERN_C_START
 #include <EXTERN.h>
 #include <perl.h>
+/* Perl 5.26.0 BOM_UTF8 clashes with WxGTK's BOM_UTF8
+ * CPAN RT#121464,
+ * fixed in newer WxGTK <http://trac.wxwidgets.org/ticket/13599>. */
+#ifdef BOM_UTF8
+#undef BOM_UTF8
+#endif
 
 #if WXPERL_P_VERSION_GE( 5, 16, 0 ) && WXPERL_P_VERSION_LT( 5, 18, 0 ) && defined(__WXOSX_COCOA__)
 #ifdef dNOOP
-- 
2.9.4

